### PR TITLE
[Core] Strategies Improvements

### PR DESCRIPTION
**Pull request by philbucher**
Hi @KratosMultiphysics/technical-committee 
The main reason for this PR is the Splitting of the `Solve` function of the Linear Strategy as requested by @RiccardoRossi in #1911 

While doing it I also fixed some more issues:
- Some methods in the Linear and the NR Strategy were private even though they are public in the Baseclass => those were moved to public again (didn't change the behavior)
- Timers were consistently replaced with the `BuiltinTimer`
- Unnecessary header-includes were removed
- some variables were renamed to match the Style Guide
- In Lin Strategy: some functions were reordered such that they are in the same order as in the BaseClass (in NR this was already the case) => only cosmetics
- The Solve function of the Lin Strategy was properly split into the subfunctions, same as what the NR does
- NR: Initialize & InitializeSolutionStep were inconsistent, this was fixed (e.g. ResizeAndInitializeVectors was called all the time)

I talked to @jcotela abt the changes but three questions remain:
- In NR: In SolveSolutionStep I believe the first check of the RebuildLevel should be checked against >0 not agains >1, see the description in the BaseClass
- In LIN: The scheme was missing the call to FinializeNonlinIteration
- NR: MoveMesh is called in Predict, I am not sure if this should be done (if yes, should it also be done in the Linear one?)

I realize that it looks like there are quite some changes but I did not really modify code (except fixing Initialize&InitializeSolutionStep for NR). I splitted the commits very clearly such that it is easier to be reviewed

Tests are running in Release and FullDebug

### CODE REVIEW

File: kratos/solving_strategies/strategies/residualbased_linear_strategy.h [hunk 1]
```diff
@@ -357,125 +382,216 @@ class ResidualBasedLinearStrategy
 
     double Solve() override
     {
-        KRATOS_TRY
```
Comment (RiccardoRossi) [inline kratos/solving_strategies/strategies/residualbased_linear_strategy.h:360]: i would put the KRATOS_TRY and KRATOS_CATCH blocks

**Review by RiccardoRossi — COMMENTED** [review]
this looks fine to me. 

anyone with comments?

File: kratos/solving_strategies/strategies/residualbased_linear_strategy.h (hunk 1 above)
Comment (philbucher) [reply kratos/solving_strategies/strategies/residualbased_linear_strategy.h:360]: @RiccardoRossi can do

Should I also add it in the BaseClass?

Comment (philbucher) [reply kratos/solving_strategies/strategies/residualbased_linear_strategy.h:360]: Ping @RiccardoRossi

**Comment by jcotela** [comment]
I am fine with the changes, but also worried that the last bullet three points might introduce unexpected issues in existing code. I'd say we give everyone one last chance to object, otherwise I'll approve this **next Monday**.

**Review by jcotela — APPROVED** [review]
Approving. Thanks @philbucher

**Pull request merged**